### ISSUE TITLE
names that are "too similar" should not display as available

### ISSUE THREAD

**Issue by chase-moskal** [remarkablemark/npm-package-name-checker#11]
npm has a mechanism to prevent publishing with names similar to existing packages.

for example, `snapstate` is shown as available in this app,

however, the name is too similar to `snap-state`, and thus cannot be published.

**Comment by remarkablemark** [comment]
You're right @chase-moskal. However, there's no easy way to determine that via the public API. This may be something that may be too complex to handle for this simple website.

**Comment by chase-moskal** [comment]
@remarkablemark, ah, that's too bad npm doesn't provide a proper api for this case.

perhaps there's a workaround -- maybe this site could query `http://registry.npmjs.com/-/v1/search?text=<searchstring>&size=20` to pull up a list that includes the similar names, and then remove all hypens from the names, to check if there is a name that matches

```js
const packageName = "snapstate"
const similarNames = await querySimilarPackageNames(packageName)
const similarNamesWithoutHyphens = similarNames.map(name => name.replace("-", ""))
const packageNameWithoutHyphens = packageName.replace("-", "")

const isTooSimilar = similarNamesWithoutHyphens.includes(packageNameWithoutHyphens)
```

just a thought

**Comment by remarkablemark** [comment]
Thanks for sharing your approach. I'll see if I have time to implement it.

**Comment by jayf0x** [comment]
Was this ever added? Had the same issue with other tools, would be nice if any of the npm checkers solved this, it's a selling point ;)

**Comment by remarkablemark** [comment]
@jayf0x unfortunately I haven't made time for this, PRs are welcome if you'd like to take a stab.
